client/src/controllers/LinkController.ts

```diff
@@ -30,6 +50,11 @@ export class LinkController extends Controller<HTMLElement> {
   }
 
   connect() {
+    // If reflectKeys is empty at the time of connect,
+    // we'll be reflecting all keys that are present in the URL.
+    if (!this.reflectKeysValue.length) {
+      this.reflectKeysValue = Array.from(new Set(this.url.searchParams.keys()));
+    }
```

Comment:
Just flagging, this will mean that you can never set the controller to an empty array.

May I suggest we change this implementation a bit, we can use the valueChanged callbacks to work out whether we should be overriding this at all by checking the second arg (the original) value.

Instead of putting this in `connect`, something like this could work.

```ts
  reflectKeysValueChanged(newValue, oldValue) {
    if (oldValue === undefined) {
      this.reflectKeysValue = Array.from(new Set(this.url.searchParams.keys()));
    }
  }
```

This will run **before** connect and also run whenever the keys value is changed by any other DOM change (which is unlikely based on the current usage).

By checking the `oldValue` (you can call this `currentValue` if you want), is undefined we can check that there is no value set on the DOM.

This way if the attribute IS set (e.g. `data-reveal-reflect-keys-value="[]"`) it will stay as an empty array, if it's not set in the DOM at all, it will then be changed to whatever params are in the element's `get url` scope.

Reply:
Ahh, thank you! I was trying to do this exact thing, but I thought this was impossible if you set a default. I didn't know that the callback is called with `undefined` as the `oldValue` if the attribute doesn't exist before `connect()`. Good to know!